subxt/src/book/usage/transactions.rs

```diff
@@ -77,40 +77,43 @@
 //! Let's see how to use each of these approaches:
 //!
 //! ```rust
-//! use subxt::config::PolkadotConfig;
-//! use std::str::FromStr;
-//!
-//! //// 1. Use a `subxt_signer` impl:
-//! use subxt_signer::{ SecretUri, sr25519 };
-//!
-//! // Get hold of a `Signer` for a test account:
-//! let alice = sr25519::dev::alice();
-//!
-//! // Or generate a keypair, here from an SURI:
-//! let uri = SecretUri::from_str("vessel ladder alter error federal sibling chat ability sun glass valve picture/0/1///Password")
-//!     .expect("valid URI");
-//! let keypair = sr25519::Keypair::from_uri(&uri)
-//!     .expect("valid keypair");
-//!
-//! //// 2. Use the corresponding `sp_core::Pair` impl:
-//! use subxt::tx::PairSigner;
-//! use sp_core::Pair;
-//!
-//! // Get hold of a `Signer` for a test account:
-//! let alice = sp_keyring::AccountKeyring::Alice.pair();
-//! let alice = PairSigner::<PolkadotConfig,_>::new(alice);
-//!
-//! // Or generate a keypair, here from an SURI:
-//! let keypair = sp_core::sr25519::Pair::from_string("vessel ladder alter error federal sibling chat ability sun glass valve picture/0/1///Password", None)
-//!     .expect("valid URI");
-//! let keypair = PairSigner::<PolkadotConfig,_>::new(keypair);
-//! #
-//! # // Test that these all impl Signer trait while we're here:
-//! #
-//! # fn is_subxt_signer(_signer: impl subxt::tx::Signer<PolkadotConfig>) {}
-//! # is_subxt_signer(subxt_signer::sr25519::dev::alice());
-//! # is_subxt_signer(subxt_signer::ecdsa::dev::alice());
-//! # is_subxt_signer(PairSigner::<PolkadotConfig,_>::new(sp_keyring::AccountKeyring::Alice.pair()));
+//! #[cfg(feature = "substrate-compat")]
+//! {
```

Comment:
Ooh, putting the feature on a block is much cleaner than my suggesiton; good idea!

I'd suggest hiding these lines though (with a `#` prefix) and not indenting the rest, since (a) most of it doesn't require that feature and (b) we talk about needing it above anyway, so keeping the example code neater feels better :)

Reply:
@tadeohepperle this is the only thing left that I'd like to see done :)